src/pages/api/signUp.ts

```diff
@@ -0,0 +1,29 @@
+import bcrypt from 'bcrypt'
+import connectDB from "@/lib/mongodb";
+import User from "@/types/User";
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+    if (req.method !== 'POST') {
+        return res.status(405).end()
+    }
+    const { name, email, password, rePassword } = req.body
+    if (password !== rePassword) {
+        return res.status(400).json({ error: 'Password miss match' })
+    }
+
+    await connectDB()
+    const isUserExist = await User.findOne({ email })
+    if (isUserExist) {
+        return res.status(409).json({ error: 'User alredy exist' })
```

Comment:
you have typo in `already`. Download sonar lint plugin for VS Code that is checking for typos. Its not good to see typos in code